common/Contracts/IDevHomeActionSetRender.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using AdaptiveCards.Rendering.WinUI3;
+
+namespace DevHome.Common.Contracts;
+
+/// <summary>
+/// Represents a renderer for an adaptive card action set that Dev Home can use to invoke actions from within
+/// the action set. This is useful for invoking an adaptive card action from an abitraty button within Dev Home's UI.
+/// </summary>
+public interface IDevHomeActionSetRender : IAdaptiveElementRenderer
```

Comment:
I probably just don't understand, but: I feel like what you want is an Action. Then you can either use the action on its own, or put it in an ActionSet. Right?

Reply:
I guess to me it felt more natural to have a custom action set, but allow extension developers to use regular adaptive card actions within it in their Json. There definitely is an opportunity to add custom actions in the future. But for this change as long as an action set is provided we can stitch up as many buttons as we want inside of it. I chose the custom action set for my use case because I wanted the next and previous buttons to be apart of a single set and not separate.